ui/app/ducks/metamask/metamask.js

```diff
@@ -154,9 +154,26 @@ function reduceMetamask (state, action) {
       return newState
 
     case actions.SET_SELECTED_TOKEN:
-      return extend(metamaskState, {
+      newState = extend(metamaskState, {
         selectedTokenAddress: action.value,
       })
+      const newSend = extend(metamaskState.send)
+
+      if (metamaskState.send.editingTransactionId && !action.value) {
```

Comment:
This is meant to handle the specific case of when we are editing a transaction and ETH is selected, correct?

I see in `send-asset-row.component` that we pass an empty string when ETH is selected. Perhaps instead of just `!action.value`, we could explicitly check `action.value === ''`. That way the reader of the code here knows that the value passed to the action was intentionally an empty string, and not just defaulting to undefined.

Reply:
changed this to `this.setSelectedToken()` to be consistent with how `setSelectedToken` is called in other area of the app when de-selecting a token

Reply:
changed it to `this.setSelectedToken()` to invoke with no params, keeping it consistent with how it is call everywhere else in the app.